task-worklow-refactor/README.md

```diff
@@ -0,0 +1,94 @@
+# How to split one Celery task into multiple tasks
+
+Let's talk about splitting the `task.process_message` into one processing task
+and multiple tasks handling the specific events. The main benefit from this 
+will be the possibility to use more queues (for faster/slower tasks)
+ and to differentiate tasks easily (Sentry).
+
+
+## Workflow
+ - get webhook payload or message from message bus 
+ - add aditional info which can make determining the event type easier
+ (GH webhooks - X-GitHub-Event)
+ - send it to Celery as `task.process_message` - this task will take care about:
+    - parsing of the event into object of class Event (this class provides `project`, `package_config` properties,
+     which enables us to make the checks)
+    - private repository check
+    - getting the handlers which handle the event
+    - whitelist check
+    - possibly creating records in DB (explained below)
+    - (running `pre-check` here to filter out some events?)
```

Comment:
Good point. We have `pre-check` both for Events and Handlers. It makes sense to run them before creating the new task.